bgpd/bgp_evpn.c

```diff
@@ -583,13 +619,20 @@ static int bgp_zebra_send_remote_macip(struct bgp *bgp, struct bgpevpn *vpn,
 		stream_putl(s, ipa_len);
 		stream_put(s, &p->prefix.macip_addr.ip.ip.addr, ipa_len);
 	}
-	stream_put_in_addr(s, &remote_vtep_ip);
+	/* tape out the VTEP-IP if the ESI is non-zero to avoid incorrect
+	 * mods
+	 */
+	if (memcmp(zero_esi, esi, sizeof(esi_t)))
```

Comment:
This doesn't look right - I expect it must be fixed by a later commit right?

Reply:
This code has morphed a bit; but the logic is the same. This is what the final code looks like -
` /* If the ESI is valid that becomes the nexthop; tape out the
     * VTEP-IP for that case
     */
    if (bgp_evpn_is_esi_valid(esi))
        stream_put_in_addr(s, &zero_remote_vtep_ip);
    else
        stream_put_in_addr(s, &remote_vtep_ip);
`

With - 
`static inline bool bgp_evpn_is_esi_valid(esi_t *esi)
{
    return !!memcmp(esi, zero_esi, sizeof(esi_t));
}`

zero_esi is a global i.e. implicitly inited to zero and used in places where ESI is not applicable.